vault/token_store.go

```diff
@@ -88,6 +95,75 @@ func NewTokenStore(c *Core, config *logical.BackendConfig) (*TokenStore, error)
 
 		Paths: []*framework.Path{
 			&framework.Path{
+				Pattern: "roles/?$",
+
+				Callbacks: map[logical.Operation]framework.OperationFunc{
+					logical.ListOperation: t.tokenStoreRoleList,
+				},
+
+				HelpSynopsis:    tokenListRolesHelp,
+				HelpDescription: tokenListRolesHelp,
+			},
+
+			&framework.Path{
+				Pattern: "roles/" + framework.GenericNameRegex("role_name"),
+				Fields: map[string]*framework.FieldSchema{
+					"role_name": &framework.FieldSchema{
+						Type:        framework.TypeString,
+						Description: "Name of the role",
+					},
+
+					"allowed_policies": &framework.FieldSchema{
+						Type:    framework.TypeString,
+						Default: "",
+						Description: `If set, tokens created via this role
```

Comment:
We can. The benefit to having it here is to attempt to make whitespacing sane across multiple options, although ideally we just need to make a better output formatter.